<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
 <TITLE>31157</TITLE>
<META http-equiv="Content-Type" content="text/html; charset=UTF-8">
  <script type="text/javascript">
  <!--
   function ZweiFrames(URL1,F1,URL2,F2)
   {
    parent.frames[F1].location.href=URL1;
    parent.frames[F2].location.href=URL2;
   }
  //-->
  </script>
</HEAD>
<BODY BGCOLOR="#ffffff" style="margin-left:25">
<HR>
<H3><CENTER>Week8_Task2.java</CENTER></H3><HR>
<PRE>
<A NAME="0"></A><FONT color="#0000ff"><div style="position:absolute;left:0"><A HREF="javascript:ZweiFrames('match61193-0.html#0',2,'match61193-top.html#0',1)"><IMG SRC="back.gif" ALT="other" BORDER="0" ALIGN="left"></A></div><B>import java.io.FileNotFoundException;
import java.io.IOException;

public class Week8_Task2 {
    public void NPEx() {
        throw new NullPointerException(&quot;Lỗi Null Pointer&quot;);
    }
    public String NPExTest(){
        String res=&quot;Không có lỗi&quot;;
        try{
            NPEx();
        }
        catch (Exception x){
            res=x.getMessage();
        }
        return res;
    }

    public void AIOoBEx() {
        throw new ArrayIndexOutOfBoundsException(&quot;Lỗi Array Index Out of Bounds&quot;);
    }
    public String AIOoBExTest(){
        String res=&quot;Không có lỗi&quot;;
        try{
            AIOoBEx();
        }
        catch (Exception x){
            res=x.getMessage();
        }
        return res;
    }

    public void ArithEx() {
        throw new ArithmeticException(&quot;Lỗi Arithmetic&quot;);
    }
    public String ArithExTest(){
        String res=&quot;Không có lỗi&quot;;
        try{
            ArithEx();
        }
        catch (Exception x){
            res=x.getMessage();
        }
        return res;
    }

    public void FNFEx() throws FileNotFoundException {
        throw new FileNotFoundException(&quot;Lỗi File Not Found&quot;);
    }
    public String FNFExTest(){
        String res=&quot;Không có lỗi&quot;;
        try{
            FNFEx();
        }
        catch (Exception x){
            res=x.getMessage();
        }
        return res;
    }

    public void IOEx() throws IOException {
        throw new IOException(&quot;Lỗi IO&quot;);
    }
    public String IOExTest(){
        String res=&quot;Không có lỗi&quot;;
        try{
            IOEx();
        }
        catch (Exception x){
            res=x.getMessage();
        }
        return res;
    }


}</B></FONT>
</PRE>

</BODY>
</HTML>
